lib/smack/SmackInstGenerator.cpp

```diff
@@ -144,6 +144,38 @@ void SmackInstGenerator::processInstruction(llvm::Instruction& inst) {
   nextInst++;
 }
 
+void SmackInstGenerator::processIntrinsicCall(llvm::IntrinsicInst* ii) {
+  static const auto f16UpCast = [&ii, this] {
+    std::string funcName;
+    auto tyID = ii->getFunctionType()->getReturnType()->getTypeID();
+    if (tyID == llvm::Type::FloatTyID)
+      funcName = "__SMACK_convert_from_fp16_f32";
+    else if (tyID == llvm::Type::DoubleTyID)
+      funcName = "__SMACK_convert_from_fp16_f64";
+    return Stmt::call(funcName, {rep->expr(ii->getArgOperand(0))},
+      {naming->get(*ii)});
+  };
+  static const auto f16DownCast = [&ii, this] {
+    std::string funcName;
+    auto tyID = ii->getArgOperand(0)->getType()->getTypeID();
+    if (tyID == llvm::Type::FloatTyID)
+      funcName = "__SMACK_convert_to_fp16_f32";
+    else if (tyID == llvm::Type::DoubleTyID)
+      funcName = "__SMACK_convert_to_fp16_f64";
+    return Stmt::call(funcName, {rep->expr(ii->getArgOperand(0))},
```

Comment:
What happens if neither `if` nor `else if` are executed?
I suspect we want to add here as a sanity check in future LLVM versions:
```
else
  llvm_unreachable("Unexpected operand type of half-precision downcast intrinsic.");
```

Reply:
Same as the previous comment.